src/pages/dashboardTab/store/dashboardReducer.js

```diff
@@ -351,6 +358,86 @@ export async function fetchAllFileIDsForSelectAll(fileCount = 100000) {
   return filteredFilesArray;
 }
 
+/**
+ * Gets all file ids for active subjectIds.
+ * TODO this  functtion can use filtered file IDs except for initial load
+ * @param obj fileCoubt
+ * @return {json}
+ */
+export async function fetchAllFileIDs(fileCount = 100000, selectedIds = [], offset = 0.0, first = 100000, order_by = 'file_name') {
+  let filesArray = [];
+
+  // If in file tab and get file name then, fetch file id from backend.
+  if (getState().currentActiveTab === tabIndex[2].title) {
+    filesArray = await client
+      .query({
+        query: GET_FILE_IDS_FROM_FILE_NAME,
+        variables: {
+          file_name: selectedIds,
+          offset,
+          first,
+          order_by,
+        },
+      })
+      .then((result) => {
+        if (result && result.data && result.data.fileIdsFromFileNameDesc.length > 0) {
+          return result.data.fileIdsFromFileNameDesc.map((d) => d.file_uuid);
+        }
+        return [];
+      });
+  } else {
+  // if in sample or case tab, get file id from backend
+    const SELECT_ALL_QUERY = getState().currentActiveTab === tabIndex[1].title
+      ? GET_ALL_FILEIDS_SAMPLESTAB_FOR_SELECT_ALL
+      : GET_ALL_FILEIDS_CASESTAB_FOR_SELECT_ALL;
+
+    let caseIds = [];
+    let sampleIds = [];
+    if (getState().currentActiveTab === tabIndex[1].title) {
```

Comment:
Looks like are using getState().currentActiveTab multiple places and multiple times in this function can you declare it as contact at starting of function?